pkg/resourcemanager/webhook/highavailabilityconfig/handler.go

```diff
@@ -214,13 +220,19 @@ func (h *Handler) handleStatefulSet(
 		&statefulSet.Spec.Template,
 	)
 
+	enforceSpreadAcrossHosts := false
+	if b, err := strconv.ParseBool(statefulSet.Annotations[resourcesv1alpha1.HighAvailabilityConfigHostSpread]); err == nil {
+		enforceSpreadAcrossHosts = b
+	}
+
```

Comment:
Nit: You could move this into `mutateTopologySpreadConstraints` to eliminate duplicated code.

Reply:
Addressed with https://github.com/gardener/gardener/pull/8970/commits/bb0f380dbdb71823b8e2c11b86c0baf70ead763d.